### PR TITLE
feat(smn): add extension in SMN subscription

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/smn' TESTARGS='-run TestAccSMNV2Subscription_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/smn -v -run TestAccSMNV2Subscription_basic -timeout 360m -parallel 4
=== RUN   TestAccSMNV2Subscription_basic
=== PAUSE TestAccSMNV2Subscription_basic
=== CONT  TestAccSMNV2Subscription_basic
--- PASS: TestAccSMNV2Subscription_basic (32.47s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/smn       32.509s
```

### CODE REVIEW

File: docs/resources/smn_subscription.md [hunk 1]
```diff
@@ -54,6 +54,32 @@ The following arguments are supported:
...
+* `extension` - (Optional, List, ForceNew) Specifies the extension configurations.
+  The [extension](#extension) structure is documented below.
+  Changing this parameter will create a new resource.
+
+<a name="extension"></a>
+The `extension` block supports:
+
+* `client_id` - (Optional, String, ForceNew) Specifies the client ID. This field is the tenant ID field in
+  the WeLink subscription and is obtained by the tenant from WeLink. This field is mandatory when `protocol`
+  is set to **welink**. Changing this parameter will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/smn_subscription.md:73]: `welink` is not supported in protocol

File: docs/resources/smn_subscription.md [hunk 2]
```diff
@@ -54,6 +54,32 @@ The following arguments are supported:
...
+  field obtained by the tenant from WeLink. This field is mandatory when `protocol` is set to **welink**.
+  Changing this parameter will create a new resource.
+
+* `keyword` - (Optional, String, ForceNew) Specifies the keyword. When `protocol` is set to **feishu**,
+  either `keyword` or `sign_secret` must be specified. When you use `keywords` to configure a security policy
+  for the Lark or DingTalk chatbot on SMN, the keywords must have one of the keywords configured on the Lark
+  or DingTalk client. Changing this parameter will create a new resource.
+
+* `sign_secret` - (Optional, String, ForceNew) Specifies the key including signature. When `protocol` is set
+  to **feishu** or **dingding**, this field or `keyword` must be specified. The key configurations must be
```
Comment (ShiChangkuo) [inline docs/resources/smn_subscription.md:85]: ditto

File: docs/resources/smn_subscription.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/smn_subscription.md:73]: fixed

File: docs/resources/smn_subscription.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/smn_subscription.md:85]: fixed

**Pull request merged**